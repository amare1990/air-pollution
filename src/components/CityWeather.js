import React from 'react';
import uuid from 'react-uuid';
import { useSelector } from 'react-redux';
import '../styles/CityWeather.css';
import { NavLink } from 'react-router-dom';

function CityWeather() {
  const components = useSelector((state) => state.pollution.data);
  const aqi = useSelector((state) => state.pollution.main);

  const cityData = useSelector((state) => state.city.data);
  const cityName = cityData.name;

  let qualitativename = '';

  if (aqi.aqi === 1) {
    qualitativename = 'Good';
  } else if (aqi.aqi === 2) {
    qualitativename = 'Fair';
  } else if (aqi.aqi === 3) {
    qualitativename = 'Moderate';
  } else if (aqi.aqi === 4) {
    qualitativename = 'poor';
  } else if (aqi.aqi === 5) {
    qualitativename = 'Very poor';
  }

  if (aqi.aqi === undefined) {
    <h2>No city data yet</h2>;
    return (<h2>No city data yet</h2>);
  }

  return (
    <div className="details-wrapper" style={{ backgroundImage: 'url(/autumn_banner.jpg)' }}>
      <NavLink
        className="to-nav-link"
        to="/"
      >
        Home

      </NavLink>

      <div className="pollution-details">

        <div className="city-weather">

          <p className="aqi">

            <span className="circumvent">
              {' '}
              {aqi.aqi}
            </span>
            {' '}
            <span className="aqi-text">{aqi.aqi && ('(air quality index(aqi))')}</span>
            {' '}
          </p>
          <h1 className="city-name">
            {cityName}
          </h1>
          <p className="quality-text">

            {aqi.aqi ? ('Feels like') : null}
            {' '}
            {'  '}
            {' '}
            <span className="qualitative-name">
              {qualitativename}
            </span>
          </p>
          <table className="pollution-data-table">
            <thead>
              <tr>
                {components.map((component) => (
                  <th key={uuid()} className="data-item">{component[0]}</th>
                ))}
              </tr>
            </thead>

            <tbody>
              <tr className="pollution-data">
                {components.map((component) => (
                  <td key={uuid()} className="data-item ">{component[1]}</td>
                ))}
              </tr>
            </tbody>
          </table>
          <p className="temprature" />
        </div>
      </div>
    </div>
  );
}

export default CityWeather;
