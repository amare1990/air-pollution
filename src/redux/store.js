import { configureStore } from '@reduxjs/toolkit';
import pollutionReducer from './current/current';
import cityReducer from './city/city';

export default configureStore({
  reducer: {
    pollution: pollutionReducer,
    city: cityReducer,
  },
});
