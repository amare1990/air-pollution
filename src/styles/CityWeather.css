.city-weather {
  background-color: #555;
  margin-top: 20px;
  padding-top: 10px;
}

.to-nav-link {
  color: #ddd;
  text-decoration: none;
  float: left;
  padding: 3px 3px;
}

.aqi {
  color: #111;
  font-size: 20px;
}

.circumvent {
  width: 30px;
  height: 30px;
  border: 1px solid rgb(201, 38, 160);
  border-radius: 50px;
  padding: 3px 3px;
}

.aqi-text {
  font-size: 12px;
}

.pollution-data-table {
  margin: 0 auto;
  border: 1px solid #fff;
  background-color: blue;
  width: 100%;
  height: 100px;
}

.data-item {
  color: #111;
  font-size: 20px;
}

.th {
  height: 70px;
}

.quality-text {
  color: aquamarine;
}

.qualitative-name {
  color: aquamarine;
  text-decoration: underline;
  font-weight: 700;
  font-size: 24px;
}
