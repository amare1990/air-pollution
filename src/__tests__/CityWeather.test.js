import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import CityWeather from '../components/CityWeather';
import store from '../redux/store';

test('Testing if CityWeather component renders correctly', () => {
  const cityWeatherSnapshot = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <CityWeather />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(cityWeatherSnapshot).toMatchSnapshot();
});
