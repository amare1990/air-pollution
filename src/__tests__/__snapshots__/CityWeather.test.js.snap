// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing if CityWeather component renders correctly 1`] = `
<h2>
  No city data yet
</h2>
`;
